web/src/hash_util.ts

```diff
@@ -109,7 +109,7 @@ export function by_stream_topic_url(stream_id: number, topic: string): string {
 // corresponding hash: the # component
 // of the narrow URL
 export function operators_to_hash(operators?: Operator[]): string {
-    let hash = "#";
+    let hash = "#all_messages";
```

Comment:
Maybe worth a comment explaining why we're using "#all_messages" here?

Reply:
Do we need to check whether your default view is `#all_messages`? We often want to avoid spelling out the full view name when it's the default view, I think?

Reply:
This is for a narrow filtering to no search operands, which I thought was the all messages view (i.e. no filters), whereas the default view is what you get when you navigate to `/`. 

How does that understand compare to how you view searching for the empty string? It's worth noting that I'd also like to change searching for the empty string to actually not change your view at all, so this would ideally be not that relevant soon.

Reply:
It seems like it should be fine to leave this out of this PR, so I removed it for now.

Reply:
Ah okay I figured out why I added this (sorry, coming back after a week away):

If you search for empty string when in the Recent Conversations view, it goes to All Messages. If we redirect to `#` as a part of that, then browser navigation later might break when we come back since `#` might render Recent Conversations.

Reply:
We go to the All Messages view from here:

```
    if (raw_operators.length === 0) {
        deactivate(coming_from_recent_topics);
        return;
    }
```

And we go to `#` using `operators_to_hash` when we go to a new narrow. Since narrowing means going to a messages view, I think it should always go to #all_messages

Reply:
Pushed a change with a comment :)